### DISCUSSION TITLE
How to mock useParams, usePathname and useSearchParams for cypress component tests?

### DISCUSSION THREAD

**Discussion by ignaciolarranaga** [vercel/next.js#51832] [App Router]
I have some cypress component tests where the components use those hooks.
Is there any gist by chance on how to mock the context for this?

I've seen I have to mock the GlobalLayoutRouterContext, but trying to figure out how to do it.

**Comment by dewodt** [comment]
> I have some cypress component tests where the components use those hooks. Is there any gist by chance on how to mock the context for this?
> 
> I've seen I have to mock the GlobalLayoutRouterContext, but trying to figure out how to do it.

Any update on this?

**Comment by alamenai** [comment]
I face the same issue.

**Comment by philippkolbe** [comment]
How to mock `useSearchParams`  is described in [this stackoverflow post](https://stackoverflow.com/questions/78129573/how-to-mock-nextjss-usesearchparams-hook-in-cypress)
